weldx/core.py

```diff
@@ -853,7 +855,7 @@ class GenericSeries:
     represented by a variable or be part of one of the expressions parameters.
     """
 
-    _required_dimension_units: dict[str, pint.Unit] = {}
+    _required_dimension_units: dict[str, Union[str, pint.Unit]] = {}
```

Comment:
I see a lot of changes like this one from `pint.Unit` to `Union[str, pint.Unit]` for internal attributes or functions. Is there a specific reason for using `str` here (and in the other internal classes)?

Internally everything should run only with `pint.Unit`, only in user facing functions `str` makes sense in my eyes.

Reply:
I have to double-check this. My first priority was to satisfy mypy. IIRC this is correct since `_required_dimension_units` is just to set some constraints to the `GenericSeries` and we can do this by using `pint.Unit` or a unit string, which is easier to read. This is passed along between multiple functions. At some point when it is necessary, strings get converted to units, but this might happen further down the road.